### ISSUE TITLE
test: dailytest failed at 20240709

### ISSUE THREAD

**Issue by pxlxingliang** [deepmodeling/abacus-develop#4616]
### Describe the Testing Issue

At 20240709, the daily test failed:
1. the accuracy test of 006_27Fe failed with dav in intel: https://app.bohrium.dp.tech/abacustest/?request=GET%3A%2Fapplications%2Fabacustest%2Fjobs%2Fsched-abacustest-intel-dav-e59d40

2. lcao basis with intel compiler failed: https://app.bohrium.dp.tech/abacustest/?request=GET%3A%2Fapplications%2Fabacustest%2Fjobs%2Fsched-abacustest-intel-elpa-9dbc48. The error message is:
```
 Initial plane wave basis and FFT box
 ---------------------------------------------------------
 DONE(0.376352   SEC) : INIT PLANEWAVE

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 39 RUNNING AT dp-lbg-471-13745296
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================
```

3. Exx test failed for all cases: https://app.bohrium.dp.tech/abacustest/?request=GET%3A%2Fapplications%2Fabacustest%2Fjobs%2Fsched-abacustest-exx-test-12083e
![image](https://github.com/deepmodeling/abacus-develop/assets/91927439/8461a643-23d4-4b14-a213-219cdeb3fb9d)

![image](https://github.com/deepmodeling/abacus-develop/assets/91927439/77746972-dff9-4b25-af73-cfb60e871ede)

### Task list for Issue attackers (only for developers)

- [ ] Understand the testing issue described by the developer.
- [ ] Review the specific test case, expected and actual results, and any error messages.
- [ ] Identify the root cause of the test failure or issue.
- [ ] If a possible solution is suggested, evaluate its feasibility and effectiveness.
- [ ] Implement a fix for the test failure or issue, or create a new test case if needed.
- [ ] Verify that the fix resolves the testing issue and the test case passes.
- [ ] Review and update any relevant documentation, such as test plans or user guides.
- [ ] Ensure the testing issue is resolved and close the ticket.
- [ ] Share any lessons learned or best practices with the team to prevent similar issues in the future.

**Comment by pxlxingliang** [comment]
These tests are all related to the intel image. I noticed that the intel image is invalid since 0701 because of the update of intel mkl from 2023.2.0 to 2024.2.0 and induce the compile error of ABACUS, and the compiler error is fixed in commit 6f7207b93 at 0708. So we cannot sure if is the intel image induce the error of the daily test.

**Comment by pxlxingliang** [comment]
I checkout the codes to commit 89a629e669e (the previous one before intel compile fix commit), and compile abacus with intel mkl 2023.2.0, and rerun the intel-elpa 001 test, and it is also failed. This indicate the error is not from the intel image, but from one commit between 0701 and 0708.
The calculation is stopped at:
```
 START CHARGE      : atomic
 DONE(7.93999    SEC) : INIT SCF
 * * * * * *
 << Start SCF iteration.

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 19972 RUNNING AT 4d6348ffe5cf
=   KILLED BY SIGNAL: 9 (Killed)
===================================================================================
```

**Comment by pxlxingliang** [comment]
After commit #4613 merged into develop branch, the daily test at 20240711 is successful.
![image](https://github.com/deepmodeling/abacus-develop/assets/91927439/b40f4337-06e6-4a36-b521-2708b522278d)


- The Accuracy test of intel-dav and omp-intel-dav is failed for the large difference of stress/force/energy for 007 case.

https://app.bohrium.dp.tech/abacustest/?request=GET%3A%2Fapplications%2Fabacustest%2Fjobs%2Fsched-abacustest-omp-intel-dav-780d91
![image](https://github.com/deepmodeling/abacus-develop/assets/91927439/25adfc25-673e-4678-88b4-4333479b6c19)

**Comment by pxlxingliang** [comment]
I use the old intel mkl version and compile the latest abacus codes, and the energy/force/stress of 006_27Fe are same as previous results. This indicates the difference of daily test is from the update of intel mkl version, and I will re-produce the reference results.
```
                        energy  \
006_27Fe-41c3f5d -86954.620197   
006_27Fe-a339356 -86954.620197   
006_27Fe-c402208 -86954.620197   

                                                              force  \
006_27Fe-41c3f5d  [0.0, 0.0, 0.0, -0.0012847397, -0.0012847397, ...   
006_27Fe-a339356  [0.0, 0.0, 0.0, -0.0012847397, -0.0012847397, ...   
006_27Fe-c402208  [0.0, 0.0, 0.0, -0.0012847397, -0.0012847397, ...   

                                                             stress  
006_27Fe-41c3f5d  [-119.446266425, -0.0, -0.0, -0.0, -119.446266...  
006_27Fe-a339356  [-119.446266425, -0.0, -0.0, -0.0, -119.446266...  
006_27Fe-c402208  [-119.446266425, -0.0, -0.0, -0.0, -119.446266... 
```
[a.zip](https://github.com/user-attachments/files/16186007/a.zip)
